core/src/main/java/com/graphhopper/GraphHopper.java

```diff
@@ -1042,27 +1042,29 @@ public GHResponse route(GHRequest request) {
             // For example see #734
             checkIfPointsAreInBounds(points);
 
+            // TODO NOW for CH it will be overwritten by chProfile.getWeighting but now we need it for locationIndex -> #729
+            Weighting weighting = createWeighting(hints, encoder);
+
```

Comment:
I think its fine to create the `Weighting` early here, but should we not better take it from the `CHProfile`, something like

```
- request comes in
- if LM/CH: select preparation using request parameters and take Weighting from the preparation/profile
- else: create Weighting from request parameters
```

instead of

```
- request comes in
- create Weighting from request parameters (hopefully we managed to re-create the same Weighting that was used for the preparation)
- later: use request parameters to select CHProfile to select CHGraph etc. and overwrite Weighting (?)
```

Two questions here:

1) Should we support using another CH/LM Weighting for the request than it was used for the preparation ? Normally this is an error, but their might be some (rather special) use-cases for this (@michaz ?). Otherwise I think the first method is better because it automatically uses the right Weighting (or at least the one that is kept by the preparation).

2) How are we supposed to select the CHProfile/Weighting given the request ? I think as soon as we want to use 'real' configurable weightings for CH we need to select the profiles by name (#1708) and if we do we need to first select the CHProfile by name and then take the weighting from the profile (first method). Otherwise we would have to send the whole configuration with every request (makes sense for non-speedup algos, because thats the whole point of configurable weightings, but not so much for LM/CH where the idea is rather to create a (custom/configurable) preparation once and then use it for many or at least multiple requests.

Reply:
Yes, thanks! Can you have a look into the new commit if this is what you meant?

> Should we support using another CH/LM Weighting for the request than it was used for the preparation ? 

For LM this works under special conditions (that increase the edge weight). For CH this won't work at the moment but it could be required for "Core CH"?

> I think as soon as we want to use 'real' configurable weightings for CH we need to select the profiles by name (#1708)

I think this should be part of a different issue #1776 and the configuration (with which the CH preparation is done) could be picked up via the `vehicle` and/or `weighting` parameters.

Reply:
> Yes, thanks! Can you have a look into the new commit if this is what you meant?

Yes this is perfect!

> For LM this works under special conditions (that increase the edge weight). For CH this won't work at the moment but it could be required for "Core CH"?

Yes but do we have to support this via the GraphHopper/high-level API here ? Do you (still) need this for the block-core thing @michaz ? Or maybe we can keep going here and add this functionality later (so far this is not really supported either) ?

> I think this should be part of a different issue #1776 

Do you mean it should be part of #1776 or it should not ? Sure, it should not be part of *this* (#1838) issue.

> and the configuration (with which the CH preparation is done) could be picked up via the vehicle and/or weighting parameters (?)

This is a good/tough question, I'll move this question to #1708.

Reply:
> Should we support using another CH/LM Weighting for the request than it was used for the preparation

Depends on what 'support' means. I need to be able to _write code_ where I pick the Graph I want to use and the Weighting I want to use by myself. So there shouldn't be machinery which _actively prevents_ me from doing so. For example, nothing should try to information-hide a CH-Graph away from me until I can show it the proper weighting instance.

Reply:
But I don't need to be able to have this as, for example, a config file option.

Reply:
Ok I'll try to keep in mind that it should be easy enough to use another weighting together with a partially contracted graph.